consensus/utils.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// Contains some utility methods to allow creating hooked consensus structures
+// mostly for test code where only a few methods are needed.
+
+package consensus
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type CurrentHeaderFn func() *types.Header
+type HeaderRetrievalFn func(hash common.Hash, number uint64) *types.Header
+type HeaderByNumberRetrievalFn func(number uint64) *types.Header
+type BlockRetrievalFn func(hash common.Hash, number uint64) *types.Block
+
+// hookedChainReader is a tiny implementation of ChainReader based on callbacks
+// and a preset chain configuration. It's useful to avoid defining various custom
+// types where a cain reader isn't readily available.
+type hookedChainReader struct {
```

Comment:
Why? MakeChainReader is not even used at the moment, and whatever would use it in the future could just as well directly implement ChainReader.